engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/generic/BaseKafkaDelayedSourceFactory.scala

```diff
@@ -0,0 +1,26 @@
+package pl.touk.nussknacker.engine.kafka.generic
+
+import org.apache.flink.streaming.connectors.kafka.KafkaDeserializationSchema
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import pl.touk.nussknacker.engine.flink.api.process.FlinkContextInitializer
+import pl.touk.nussknacker.engine.flink.api.timestampwatermark.TimestampWatermarkHandler
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, PreparedKafkaTopic, RecordFormatter}
+import pl.touk.nussknacker.engine.kafka.source.KafkaSource
+
+trait BaseKafkaDelayedSourceFactory {
```

Comment:
IMO we should write some descriptive comment why we need this feature instead of using just delay component. I guess that it is usefull in case when user don't want to have big state with scheduled mssages and just want to have this "saved" it on kafka topic. If so, it looks like very technical optimization details and I wonder if we can done this without involving user - e.g. by some graph optimization checking if there is a sequence of nodes: source -> delay and in this case replace source by this source.